resources/testharness.js

```diff
@@ -568,7 +568,21 @@ policies and contribution forms [3].
         var test_name = name ? name : test_environment.next_default_test_name();
         var test_obj = new Test(test_name, properties);
         if (func) {
-            test_obj.step(func, test_obj, test_obj);
+            var value = test_obj.step(func, test_obj, test_obj);
+            if (value !== undefined) {
+                var msg = "Test named \"" + test_name +
+                    "\" inappropriately returned a value";
```

Comment:
I don't think it's clear what it means for the test to return a value.

Reply:
I agree! I've reworked the wording here, please let me know if it reads better.